### DISCUSSION TITLE
Something wrong about the caculation about the molecular descriptors

### DISCUSSION THREAD

**Discussion by thuwilliamwyx** [rdkit/rdkit#6151] [Q&A] [answered]
I have used rdkit.Chem.Descriptors to calculate the molecular descriptors 2 days ago, and get 208 descriptors using Descriptors._descList. But today there are only 123 descriptors, which are the first 123 of the previous 207 descriptors. 
I use the same codes and wonder why is this happen?

**Comment by thuwilliamwyx** [comment]
des_list = [x[0] for x in Descriptors._descList]
print(len(des_list))
this is the main codes
I use the rdkit 2021.3.5

**Comment by greglandrum** [comment, marked as answer]
This normally happens because the code cannot find the data file which contains the definitions of the fragment descriptors.
What does this command display for you?
```
python -c 'from rdkit import RDConfig;print(RDConfig.RDDataDir);from rdkit.Chem import Descriptors;print(len(Descriptors._descList))'
```

**Reply by thuwilliamwyx** [reply]
I have just try this code and find the RDConfig.RDDataDir gives the wrong conda environment path, which is not the interpreter I use. I have change the environment's name by the true project's name.

**Reply by thuwilliamwyx** [reply]
thank you, I have fixed it. Is that means I must have a environment names my-rdkit-env? It sames that the other envs which have rdkit also get data from the env my-rdkit-env, so I rename a env as my-rdkit-env and the problem is solved.

**Reply by greglandrum** [reply]
your environments can be named whatever you like.
Is it possible that you have the evironment variable RDBASE defined? If you do this (and you don't need to do it if you are using conda environments), then you need to be sure that RDBASE points to the conda install. Here's a demonstration of that:
```
(py310_rdkit) glandrum@Badger:~$ python -c 'from rdkit import RDConfig;print(RDConfig.RDDataDir);from rdkit.Chem import Descriptors;print(len(Descriptors._descList))'
/home/glandrum/miniconda3/envs/py310_rdkit/share/RDKit/Data

208
(py310_rdkit) glandrum@Badger:~$ 
(py310_rdkit) glandrum@Badger:~$ export RDBASE='/tmp'
(py310_rdkit) glandrum@Badger:~$ python -c 'from rdkit import RDConfig;print(RDConfig.RDDataDir);from rdkit.Chem import Descriptors;print(len(Descriptors._descList))'
/tmp/Data
123
(py310_rdkit) glandrum@Badger:~$ export RDBASE=/home/glandrum/miniconda3/envs/py310_rdkit/share/RDKit
(py310_rdkit) glandrum@Badger:~$ python -c 'from rdkit import RDConfig;print(RDConfig.RDDataDir);from rdkit.Chem import Descriptors;print(len(Descriptors._descList))'
/home/glandrum/miniconda3/envs/py310_rdkit/share/RDKit/Data
208
```
